cli/azd/cmd/provision.go

```diff
@@ -30,28 +31,21 @@ type provisionFlags struct {
 }
 
 func (i *provisionFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
-	i.bindNonCommon(local, global)
-	i.bindCommon(local, global)
-}
+	existing := local.Lookup("no-progress")
```

Comment:
Would love to understand the motivation here. I would love to avoid no-opts like this, because it may lead to uncaught errors when a flag is incorrectly registered twice. It means that it's possible the values parsed will go to a different `flag` instance and not the current one.

It's better to error out, and avoid registering duplicate flags by construction (why would a flag need to be registered twice?)

Reply:
I would love to have the `flags` changes be tackled as a separate PR to limit scope.